rest-client/src/main/java/io/apicurio/registry/client/RegistryRestClientImpl.java

```diff
@@ -138,6 +140,8 @@ private static OkHttpClient createHttpClientWithConfig(String baseUrl, Map<Strin
             if (user != null && !user.isEmpty()) {
                 String credentials = Credentials.basic(user, pwd);
                 requestHeaders.put("Authorization", credentials);
+            } else if (auth != null) {
+                okHttpClientBuilder.addInterceptor(new AuthInterceptor(auth));
```

Comment:
This means that if the URL's username/password takes precedence over the inclusion of an `Auth`.  I'm OK with it if that was your intention.  But I could see an argument that an `Auth` is more explicit, programmatically, and that it should take precedent over any user/pass info in the URL.  Thoughts?

Reply:
Also what do you think about implementing a `BasicAuth` class to use.  So the code could change like this:

```java
if (auth == null && user != null && !user.isEmpty()) {
    auth = new BasicAuth(user, pwd);
}
if (auth != null) {
    okHttpClientBuilder.addInterceptor(new AuthInterceptor(auth));
}
```
It also allows users to directly use the convenient `BasicAuth` class to provide basic credentials if they prefer to do that instead of the URL method.